src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import duke.commands.AddDeadlineCommand;
+import duke.commands.AddEventCommand;
+import duke.commands.AddTodoCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.InvalidCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+
+import duke.exceptions.NotEnoughArgumentsException;
+import duke.exceptions.TaskNoDescriptionException;
+import duke.exceptions.DateNotInSequenceException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This is a class responsible for parsing the inputs given by user
+ */
+public class Parser {
+
+    /**
+     * Constructor for Parser class
+     */
+    public Parser() {}
+
+    /**
+     * Returns a Command corresponding to the information given by the user
+     *
+     * @param fullCommand information given by the user
+     * @return corresponding Command
+     */
+    public static Command parse(String fullCommand) {
+        String[] commandParts = fullCommand.split(" ", 2);
+        String commandHeader = commandParts[0];
+        switch (commandHeader) {
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+            case "todo":
+                return parseTodoCommand(fullCommand);
+            case "deadline":
+                return parseDeadlineCommand(fullCommand);
+            case "event":
+                return parseEventCommand(fullCommand);
+            case "mark":
+                return parseMarkCommand(fullCommand);
+            case "unmark":
+                return parseUnmarkCommand(fullCommand);
+            case "delete":
+                return parseDeleteCommand(fullCommand);
+            case "find":
+                return parseFindCommand(fullCommand);
+            default:
+                return new InvalidCommand();
+        }
+    }
+
+    private static AddTodoCommand parseTodoCommand(String fullCommand) throws TaskNoDescriptionException {
+        String info = fullCommand.substring(4).trim();
+        if (info.isEmpty()) {
+            throw (new TaskNoDescriptionException("OOPS!!! The description of a todo cannot be empty."));
+        }
+
+        return new AddTodoCommand(info);
+    }
+
+    private static AddDeadlineCommand parseDeadlineCommand(String fullCommand)
+        throws TaskNoDescriptionException, NotEnoughArgumentsException, DateTimeParseException {
+        String info = fullCommand.substring(8).trim();
+        if (info.isEmpty()) {
+            throw(new TaskNoDescriptionException("OOPS!!! The description of a deadline cannot be empty."));
```

Comment:
I have noticed that you put parentheses around new thrown exceptions. I don't think that is against the coding standards, but perhaps leave a space between `throw` and `(`. This is because `throw` is a Java reserved word, so should be followed by a white space.